ekm/slashing_protection_storage.go

```diff
@@ -0,0 +1,185 @@
+package ekm
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/bloxapp/eth2-key-manager/core"
+	ssz "github.com/ferranbt/fastssz"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/logging"
+	"github.com/bloxapp/ssv/storage/basedb"
+)
+
+const (
+	highestAttPrefix      = prefix + "highest_att-"
+	highestProposalPrefix = prefix + "highest_prop-"
+
+	// versionPrefix is the version prefix key of slashing protection DB
+	versionPrefix = "version"
+	// GenesisVersion is the genesis version of the slashing protection DB
+	GenesisVersion = "0x0"
+)
+
+type SpStorage interface {
+	core.SlashingStore
+
+	RemoveHighestAttestation(pubKey []byte) error
+	RemoveHighestProposal(pubKey []byte) error
+
+	// SetVersion saves the version of the slashing protection DB
+	SetVersion(version string) error
+	// GetVersion returns the version of the slashing protection DB
+	GetVersion() (string, bool, error)
+}
+
+type spStorage struct {
+	db     basedb.Database
+	logger *zap.Logger // struct logger is used because core.Storage does not support passing a logger
+	lock   sync.RWMutex
+
+	prefix []byte
+}
+
+func (s *spStorage) SetVersion(version string) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	return s.db.Set([]byte(versionPrefix), nil, []byte(version))
+}
+
+func (s *spStorage) GetVersion() (string, bool, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	obj, found, err := s.db.Get([]byte(versionPrefix), nil)
+	if err != nil {
+		return "", found, errors.Wrap(err, "failed to get version")
+	}
+	if !found {
+		return "", found, nil
+	}
+
+	return string(obj.Value), found, nil
+}
+
+func NewSlashingProtectionStorage(db basedb.Database, logger *zap.Logger, prefix []byte) SpStorage {
```

Comment:
I don't see a point returning the interface here. you could just make the struct be public, and set any functions you don't want used as private.

Reply:
pls see [comment](https://github.com/bloxapp/ssv/pull/1146/files/465528b438cf08d623bcf426fa06195dc0e19610#r1342156357)
in addition to my previous comment I think it may offer more level of abstraction and testability.
But still if u insist we can change it